api/src/main/java/com/flipkart/poseidon/oas/SchemaGenerator.java

```diff
@@ -0,0 +1,466 @@
+        final Path genDir = Paths.get(args[1]);
+        final Path apiDir = genDir.resolve("apis/v1");
+        final Path apiFile = apiDir.resolve("ServiceResource.json");
+        final Path modelsDir = genDir.resolve("models");
+
+        try {
+            APIManager.scanAndAdd(dir, validConfigs);
+            for (String config : validConfigs) {
+                pojos.add(getMapper().readValue(config, EndpointPOJO.class));
+            }
+        } catch (Exception e) {
+            throw new UnsupportedOperationException("Error while reading EndpointPOJO", e);
+        }
+
+        fetchDataSources(args[2].split(","));
+
+        OpenAPI openAPI = new OpenAPI();
+        Info info = new Info();
+        info.title("ServiceResource");
+        info.version("1.0");
+        openAPI.info(info);
+
+        final Map<String, Map<PathItem.HttpMethod, Operation>> paths = new LinkedHashMap<>();
+        for (EndpointPOJO endpointPOJO : pojos) {
+            final Operation operation = new Operation();
+            final ParamsPOJO paramsPOJO = endpointPOJO.getParams();
+            if (paramsPOJO != null) {
+                if (paramsPOJO.getRequired() != null) {
+                    for (ParamPOJO paramPOJO : paramsPOJO.getRequired()) {
+                        if (paramPOJO.isFile()) {
+                            continue;
+                        }
+
+                        if (paramPOJO.isBody()) {
+                            final RequestBody requestBody = new RequestBody();
+                            requestBody.required(true);
+                            final Content content = new Content();
+                            final MediaType mediaType = new MediaType();
+                            mediaType.schema(createSchema(paramPOJO));
+                            content.addMediaType("application/json", mediaType);
+                            requestBody.content(content);
+                            operation.requestBody(requestBody);
+                            continue;
+                        }
+
+                        operation.addParametersItem(createParameter(paramPOJO).required(true));
+                    }
+                }
+
+                if (paramsPOJO.getOptional() != null) {
+                    for (ParamPOJO paramPOJO : paramsPOJO.getOptional()) {
+                        if (paramPOJO.isFile() || paramPOJO.isBody()) {
+                            continue;
+                        }
+
+                        operation.addParametersItem(createParameter(paramPOJO));
+                    }
+                }
+            }
+
+            ApiResponses responses = new ApiResponses();
+            if (endpointPOJO.getResponse() instanceof String) {
+                String responseString = (String) endpointPOJO.getResponse();
+                responseString = responseString.substring(3, responseString.length() - 2);
+                if (responseString.startsWith("$")) {
```

Comment:
So the above two lines ensures optional starting with # (pound).

Reply:
yep